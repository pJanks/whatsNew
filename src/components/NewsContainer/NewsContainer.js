import React from 'react';
import './NewsContainer.css'


const NewsContainer = (props) => {
  return props.ideas.map(newsArticle => {
    return (<article key={newsArticle.id}>
      <h2>{newsArticle.headline}</h2>
      <img src={newsArticle.img}/>
      <p>{newsArticle.description}</p>
    </article>)
  })
}

export default NewsContainer;
