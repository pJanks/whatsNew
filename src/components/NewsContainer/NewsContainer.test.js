import React from 'react';
import ReactDOM from 'react-dom';
import NewsContainer from './NewsContainer';
import { shallow } from 'enzyme';

describe('NewsContainer', () => {
  let wrapper, instance, mockNewsArticles;

  beforeEach(() => {
    mockNewsArticles = [{
      id: 3,
      headline: 'crazy news',
      img: 'aPicture',
      description: 'something crazy',
      url: 'www.hi.com'
    }, {
      id: 4,
      headline: 'more crazy news',
      img: 'anotherPicture',
      description: 'something even crazier',
      url: 'www.hello.com'
    }]
    wrapper = shallow(<NewsContainer news={mockNewsArticles} />)
    instance = wrapper.instance()
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  });
});
