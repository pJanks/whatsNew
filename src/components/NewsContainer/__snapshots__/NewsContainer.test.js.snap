// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot 1`] = `
<section
  className="news-container"
>
  <NewsArticle
    article={
      Object {
        "description": "something crazy",
        "headline": "crazy news",
        "id": 3,
        "img": "aPicture",
        "url": "www.hi.com",
      }
    }
    key="3"
  />
  <NewsArticle
    article={
      Object {
        "description": "something even crazier",
        "headline": "more crazy news",
        "id": 4,
        "img": "anotherPicture",
        "url": "www.hello.com",
      }
    }
    key="4"
  />
</section>
`;
